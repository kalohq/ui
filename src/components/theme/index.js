/* @flow */
export {default} from './theme';
