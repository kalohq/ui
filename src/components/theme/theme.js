/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
