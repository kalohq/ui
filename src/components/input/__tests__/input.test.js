/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from '../../theme';
import Input, {InputAddon} from '../input';

describe('Input', () => {
  const create = props => shallow(<Input {...props} />);

  const defaultProps = {
    size: 'medium',
    theme: 'default',
  };

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Input {...defaultProps} />
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render prefixed addon content', () => {
    const element = create({
      theme: 'default',
      addonPrefix: '£',
    });
    expect(element.find(InputAddon).length).toBe(1);
  });

  test('should render suffixed addon content', () => {
    const element = create({
      theme: 'default',
      addonSuffix: '£',
    });
    expect(element.find(InputAddon).length).toBe(1);
  });
});
