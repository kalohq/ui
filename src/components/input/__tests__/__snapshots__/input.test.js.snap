// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render shallow component ok 1`] = `
<div
  className="e6fqpax1 css-15tsz50"
  style={Object {}}
>
  <input
    className="css-1nv6z25 e6fqpax2"
    defaultValue={undefined}
    name={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    size="medium"
    style={
      Object {
        "width": undefined,
      }
    }
    value={undefined}
  />
</div>
`;
